### ISSUE TITLE
Possible bug: Ensure we correctly handle the level intermediate_parent

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/r2dii.match#84]
Relates to #31 and #32 

`match_name()` outputs values of "intermediate_parent*" under `level` but I suspect we still don't handle this level correctly. 

In `restructure_loanbook_for_matching()` I see that we uniquify `id_direct_loantaker` with prefix = "C" and `id_ultimate_parent` with prefix = "UP". We do not uniquify `id_intermediate_parent_[1-n]` and we probably should do so with prefix "IP".

Also, the column names and prefixes are hard-coded and instead with likely need to get them from available column names of the input loanbook. 

`restructure_loanbook_for_matching()` looses intermediate_parent colums at `input_cols_for_prepare_loanbook()` -- which should be fixed.

@jdhoffa, can you confirm if you also see this as something I need to fix?

**Comment by jdhoffa** [comment]
@maurolepore the reason I did not uniquify the intermediate parents was due to the potential for multiple levels (ie. the first element of intermediate_parent_1 would be IP1, and the first element of intermediate_parent_2 would also be IP1) we could potentially get around this by having prefixes like IPONE1, IPTWO1 or something similar, but this seems clunky and uncessary. 

I actually think the prefixes are unnecessary/ redundant in general. (UP1 is identifiable simply by ID and level, so explicitly stating the level isn't entirely necessary).
